### ISSUE TITLE
Ability to flag styles as NSFW

### ISSUE THREAD

**Issue by detunedradios** [OpenUserCSS/openusercss.org#144]
At the moment, the front page of openusercss.org presents me with a style of a Japanese adult video website with a thumbnail of a busty woman in lingerie. Nothing against that, of course - actually, I'm glad somewhere is willing to host userstyles for adult sites! But for a variety of reasons (public browsing, SEO, etc.) it would be useful to be able to filter out NSFW styles on the site. It may also be best to filter these out by default.

This should be as simple as a "NSFW" checkbox when uploading or editing a style, along with filters in search results. I'd probably suggest that NSFW styles either be excluded from appearing on the front page, or at least have the thumbnail blurred or mosaic'd in some way unless the user signs in and disables NSFW filtering. Moderation-wise, any style that either modifies an adult website or itself contains NSFW content (say it injects a NSFW background into your Google results) should be expected to be flagged.

**Comment by NickServ** [comment]
I agree that a NSFW setting should be available for the reasons you mentioned, but I do not think signing up should be necessary to view NSFW styles. To require visitors to sign up in order to view some content on the site would be a hassle hurdle and a privacy concern to some. This applies even if just the screenshots are blocked and not the style itself.
